generate.py

```diff
@@ -38,16 +40,16 @@ def writefile(filename, s):
     bn = basename(sys.argv[0])
     s = '// DO NOT EDIT - autogenerated by ' + bn + '\n\n' + s
     if os.path.exists(filename) and open(filename).read() == s:
-        print 'Checked', filename
+        print('Checked', filename)
     else:
-        print 'Writing', filename
+        print('Writing', filename)
         open(filename, 'w').write(s)
 
 
 def generate_enum(enum_name, enum_prefix, count_name, name_list, value_list):
     # Can only generate an enum which starts at 0, and where each value is 1 more than the former
     assert len(name_list) == len(value_list)
-    assert value_list == list(xrange(0, len(value_list)))
+    assert value_list == list(range(0, len(value_list)))
```

Comment:
Given that this gives an import error here (on Python 2), I don't think we should use it ;):

```
Python 2.7.11 (default, Mar 31 2016, 20:46:51) 
[GCC 5.3.1 20151207 (Red Hat 5.3.1-2)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from builtins import range
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named builtins
>>>
```

Both Python 2 and 3 have a `range()` function in the global namespace, the Python 2 version returns a list and the  Python 3 version returns a range iterator. Casting either to a list will do the right thing here, with only slight overhead.